### PR TITLE
Add basic auth support to MesosSchedulerClient

### PR DESCRIPTION
**Pull request by BenWhitehead**
Should merge after #23

### CODE REVIEW

File: mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java [hunk 1]
```diff
@@ -73,17 +84,35 @@ public MesosSchedulerClient(
...
+
         createPost = (Send s) -> {
             final byte[] bytes = sendCodec.encode(s);
-            return just(HttpClientRequest.createPost("/api/v1/scheduler")
+            HttpClientRequest<ByteBuf> request = HttpClientRequest.createPost(mesosUri.getPath())
+                .withHeader("User-Agent", String.format("%s %s", name, userAgent.toString()))
                 .withHeader("Content-Type", sendCodec.mediaType())
                 .withHeader("Accept", receiveCodec.mediaType())
-                .withContent(bytes)
+                .withHeader("Accept", receiveCodec.mediaType());
```
Comment (cruhland) [inline mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java]: Looks like a duplicate line.

File: mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java [hunk 2]
```diff
@@ -73,17 +84,35 @@ public MesosSchedulerClient(
...
+
+            final String userInfo = mesosUri.getUserInfo();
+            if (userInfo != null) {
+                //Won't actually work until https://issues.apache.org/jira/browse/MESOS-3923 is fixed
```
Comment (cruhland) [inline mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java]: File an issue in this repo to follow up?

File: mesos-rxjava-core/src/test/java/org/apache/mesos/rx/java/MesosSchedulerClientTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        final HttpClientRequest<ByteBuf> request = client.createPost.call(ACK)
+            .toBlocking()
+            .first();
+
+        final Map<String, String> headers = headersToMap(request.getHeaders());
+        assertThat(headers).containsKeys("Authorization");
+        final String authorization = headers.get("Authorization");
+        assertThat(authorization).isEqualTo("Basic dGVzdHVzZXI6dGVzdHBhc3N3b3Jk");
+
+        final String base64UserPass = authorization.substring("Begin ".length());
```
Comment (cruhland) [inline mesos-rxjava-core/src/test/java/org/apache/mesos/rx/java/MesosSchedulerClientTest.java]: I'm guessing you meant `"Basic ".length()` here.

File: mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java (hunk 2 above)
Comment (BenWhitehead) [inline mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java]: Done #26

**Pull request merged**